Library/src/Systems/Organizations/OrganizationSystem.cpp

```diff
@@ -129,6 +123,139 @@ OrganizationSystem::~OrganizationSystem()
+	const auto* UserSystem		= csp::systems::SystemsManager::Get().GetUserSystem();
+	const auto& OrganizationIds = UserSystem->GetLoginState().OrganizationIds;
+
+	if (OrganizationIds.Size() == 0)
+	{
+		CSP_LOG_ERROR_MSG("Unable to get current Orgaization Id, you do not belong to an Organization.");
+
+		static const String EmptyId = "";
+		return EmptyId;
+	}
+
+	return OrganizationIds[0];
+}
+
+void OrganizationSystem::CreateOrganization(const csp::common::String& OrganizationOwnerId,
+											const csp::common::String& OrganizationName,
+											OrganizationResultCallback Callback)
+{
+	auto OrganizationInfo = std::make_shared<chs::OrganizationDto>();
+	OrganizationInfo->SetName(OrganizationName);
+	OrganizationInfo->SetOrganizationOwnerId(OrganizationOwnerId);
+
+	csp::services::ResponseHandlerPtr ResponseHandler
+		= OrganizationApi->CreateHandler<OrganizationResultCallback, OrganizationResult, void, chs::OrganizationDto>(Callback, nullptr);
+
+	static_cast<chs::OrganizationApi*>(OrganizationApi)->apiV1OrganizationsPost(OrganizationInfo, ResponseHandler);
+}
+
+void OrganizationSystem::UpdateOrganization(const csp::common::Optional<csp::common::String>& OrganizationId,
+											const csp::common::String& Name,
+											OrganizationResultCallback Callback)
+{
+	if (Name.IsEmpty())
+	{
+		CSP_LOG_WARN_MSG("A valid new Organization name must be specified.");
+
+		INVOKE_IF_NOT_NULL(Callback, MakeInvalid<OrganizationResult>());
+
+		return;
+	}
+
+	csp::common::String SelectedOrganizationId;
+
+	if (OrganizationId.HasValue())
+	{
+		SelectedOrganizationId = *OrganizationId;
+	}
+	else
+	{
+		SelectedOrganizationId = GetCurrentOrganizationId();
+
+		if (SelectedOrganizationId.IsEmpty())
+		{
+			INVOKE_IF_NOT_NULL(Callback, MakeInvalid<OrganizationResult>());
+
+			return;
+		}
+	}
+
+	auto OrganizationInfo = std::make_shared<chs::OrganizationDto>();
+	OrganizationInfo->SetName(Name);
+
+	auto* UserSystem   = SystemsManager::Get().GetUserSystem();
+	const auto& UserId = UserSystem->GetLoginState().UserId;
+
+	OrganizationInfo->SetOrganizationOwnerId(UserId);
+
+	csp::services::ResponseHandlerPtr ResponseHandler
+		= OrganizationApi->CreateHandler<OrganizationResultCallback, OrganizationResult, void, chs::OrganizationDto>(Callback, nullptr);
+
+	static_cast<chs::OrganizationApi*>(OrganizationApi)
+		->apiV1OrganizationsOrganizationIdPut(SelectedOrganizationId, OrganizationInfo, ResponseHandler);
+}
+
+void OrganizationSystem::DeactivateOrganization(const csp::common::Optional<csp::common::String>& OrganizationId, NullResultCallback Callback)
+{
+	csp::common::String SelectedOrganizationId;
+
+	if (OrganizationId.HasValue())
+	{
+		SelectedOrganizationId = *OrganizationId;
+	}
+	else
+	{
+		SelectedOrganizationId = GetCurrentOrganizationId();
+
+		if (SelectedOrganizationId.IsEmpty())
+		{
+			INVOKE_IF_NOT_NULL(Callback, MakeInvalid<NullResult>());
```

Comment:
Could we also print to the log here to tell the user that they attempted to deactivate an organization, but CSP did not do anything, because no ID parameters was given and the user doesn't currently belong to one?

Reply:
Added an error log.